Rename shadowed variables

<!--

NOTE: this form is for bug reports only.
For questions or troubleshooting, please post on the protobuf mailing list:
https://groups.google.com/forum/#!forum/protobuf 
Stack Overflow is also a useful if unofficial resource https://stackoverflow.com/questions/tagged/protocol-buffers
-->

**What version of protobuf and what language are you using?**
Version: main
Language: C++

**What operating system (Linux, Windows, ...) and version?**
MacOS

**What runtime / compiler are you using (e.g., python version or gcc version)**
```cpp
# Compile failed
$ bazel build --copt='-Wshadow-all' :protoc :protobuf
$ clang --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**What did you expect to see**
Shadowed variables can cause readability issues. Ensure a shadowed
variable isn't used in header files which may be used in a dependent
project that explicitly disables them.
`bazel build --copt='-Wshadow-all' :protoc :protobuf` executed successfully.

Thanks for reporting this issue! I was able to successfully reproduce it and I see errors with shadowed variables.

For future record, here is an example of the logs:
```
bazel-out/k8-fastbuild/bin/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/serial_arena.h: In constructor 'google::protobuf::internal::ArenaBlock::ArenaBlock(google::protobuf::internal::ArenaBlock*, size_t)':
bazel-out/k8-fastbuild/bin/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/serial_arena.h:71:39: error: declaration of 'size' shadows a member of 'google::protobuf::internal::ArenaBlock' [-Werror=shadow]
   71 |   ArenaBlock(ArenaBlock* next, size_t size)
      |                                ~~~~~~~^~~~
bazel-out/k8-fastbuild/bin/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/serial_arena.h:86:16: note: shadowed declaration is here
   86 |   const size_t size;
      |                ^~~~
```

--- a/csharp/CHANGES.txt
+++ b/csharp/CHANGES.txt
@@ -72,7 +72,7 @@ Features:
   custom implementation of writers with both speed and size optimizations.
 - Addition of the "Google.ProtoBuffers.Serialization" assembly to support
   reading and writing messages to/from XML, JSON, IDictionary<,> and others.
-- Several performance related fixes and tweeks
+- Several performance related fixes and tweaks
 - Issue 3:	Add option to mark generated code with attribute
 - Issue 20:	Support for decorating classes [Serializable]
 - Issue 21:	Decorate fields with [deprecated=true] as [System.Obsolete]

--- a/src/google/protobuf/port_def.inc
+++ b/src/google/protobuf/port_def.inc
@@ -1010,6 +1010,10 @@ static_assert(PROTOBUF_ABSL_MIN(20230125, 3),
 // This error has been generally flaky, but we need to disable it specifically
 // to fix https://github.com/protocolbuffers/protobuf/issues/12313
 #pragma clang diagnostic ignored "-Wunused-parameter"
+// -Wshadow means that declaration shadows a variable.
+// for (int i = 0; i < file_->public_dependency_count(); ++i) 
+//   for (int i = 0; i < public_dep->message_type_count(); ++i)
+#pragma clang diagnostic ignored "-Wshadow"
 #endif
 #ifdef __GNUC__
 #pragma GCC diagnostic push
@@ -1033,6 +1037,10 @@ static_assert(PROTOBUF_ABSL_MIN(20230125, 3),
 // This error has been generally flaky, but we need to disable it specifically
 // to fix https://github.com/protocolbuffers/protobuf/issues/12313
 #pragma GCC diagnostic ignored "-Wunused-parameter"
+// -Wshadow means that declaration shadows a variable.
+// for (int i = 0; i < file_->public_dependency_count(); ++i) 
+//   for (int i = 0; i < public_dep->message_type_count(); ++i)
+#pragma GCC diagnostic ignored "-Wshadow"
 #ifndef __clang__
 // This causes spurious warnings in GCC 13.
 #pragma GCC diagnostic ignored "-Wstringop-overflow"

